### PR TITLE
Potentially fix a memory leak in FlairStore

### PR DESCRIPTION
**Pull request by lukebarnard1**
For each successful request of a group profile, we previously
emitted an `updateGroupProfile` event per caller of
`getGroupProfileCached`. This is sub-optimal because only a single
event emit is required to update the views listening.

It's possible that this was enabling some race to cause a memory
leak but this is not certain, hence the extra logging for future
debugging.

### CODE REVIEW

File: src/stores/FlairStore.js [hunk 1]
```diff
@@ -154,16 +154,26 @@ class FlairStore extends EventEmitter {
...
-        // No request yet, start one
-        if (!this._groupProfilesPromise[groupId]) {
-            this._groupProfilesPromise[groupId] = matrixClient.getGroupProfile(groupId);
+        // A request is ongoing, wait for it to complete and return the group profile.
+        if (this._groupProfilesPromise[groupId]) {
+            try {
+                await this._groupProfilesPromise[groupId];
+            } catch (e) {
+                // Don't log the error; this is done below
+                return undefined;
```
Comment (dbkr) [inline src/stores/FlairStore.js]: why undefined rather than null?

File: src/stores/FlairStore.js [hunk 2]
```diff
@@ -154,16 +154,26 @@ class FlairStore extends EventEmitter {
...
+            }
+            return this._groupProfiles[groupId];
         }
 
+        // No request yet, start one
+        console.log('FlairStore: Request group profile of ' + groupId);
+        this._groupProfilesPromise[groupId] = matrixClient.getGroupProfile(groupId).delay(5000);
```
Comment (dbkr) [inline src/stores/FlairStore.js]: why the delay?

Comment (lukebarnard1) [reply src/stores/FlairStore.js]: Oops, that shouldn't be there, I was attempting to exacerbate the race.

File: src/stores/FlairStore.js (hunk 1 above)
Comment (lukebarnard1) [reply src/stores/FlairStore.js]: No reason, I shall change to `null`.

**Pull request merged**